### PR TITLE
net: Improve and document SOCKS code

### PR DESCRIPTION
**Pull request by laanwj**
Make the SOCKS code more consistent, and document the constants used.

### CODE REVIEW

File: src/netbase.cpp [hunk 1]
```diff
@@ -203,15 +245,15 @@ enum class IntrRecvError {
...
+static IntrRecvError InterruptibleRecv(uint8_t* data, size_t len, int timeout, const SOCKET& hSocket)
 {
     int64_t curTime = GetTimeMillis();
     int64_t endTime = curTime + timeout;
     // Maximum time to wait in one select call. It will take up until this time (in millis)
     // to break off in case of an interruption.
     const int64_t maxWait = 1000;
     while (len > 0 && curTime < endTime) {
-        ssize_t ret = recv(hSocket, data, len, 0); // Optimistically try the recv first
+        ssize_t ret = recv(hSocket, (void*)data, len, 0); // Optimistically try the recv first
```
Comment (promag) [inline src/netbase.cpp]: Funny, in windows 2nd arg is `char*`, that's why the build failed.

Comment (laanwj) [reply src/netbase.cpp]: Oops, thought it would be void, added the wrong cast. Should be ok now.

**Comment by theuni** [comment]
utACK 46544c60d67d8a7095e7243795d2b570ab8391e3. InterruptibleRecv goes away soon as part of the async connection work, but having the SOCKS constants defined will make those changes much easier to review. Thanks for saving me a step :)

**Comment by promag** [comment]
utACK 46544c6.

File: src/netbase.cpp [hunk 2]
```diff
@@ -184,6 +184,48 @@ struct timeval MillisToTimeval(int64_t nTimeout)
...
+/** Values defined for REP in RFC1928 */
+enum SOCKS5Reply: uint8_t {
+    SUCCEEDED = 0x00,       //! Succeeded
+    GENFAILURE = 0x01,      //! General failure
+    NOTALLOWED = 0x02,      //! Connection not allowed by ruleset
+    NETUNREACHABLE = 0x03,  //! Network unreachable
+    HOSTUNREACHABLE = 0x04, //! Network unreachable
+    CONNREFUSED = 0x05,     //! Connection refused
+    TTLEXPIRED = 0x06,      //! TTL expired
+    CMDUNSUPPORTED = 0x07,  //! Command not supported
```
Comment (promag) [inline src/netbase.cpp]: Nit, align this and above comments with below.

**Review by gmaxwell — APPROVED** [review]
ACK

**Comment by laanwj** [comment]
Re-aligned comment @promag nit 46544c6 → 22f816ef4dd57785627d801c247714cace7fc5b5

**Review by gmaxwell — APPROVED** [review]
reACK

**Comment by sipa** [comment]
utACK 22f816ef4dd57785627d801c247714cace7fc5b5

**Comment by TheBlueMatt** [comment]
utACK 22f816ef4dd57785627d801c247714cace7fc5b5

**Comment by jonasschnelli** [comment]
utACK 22f816ef4dd57785627d801c247714cace7fc5b5

**Comment by jonasschnelli** [comment]
Tested ACK (macOS 10.12) 22f816ef4dd57785627d801c247714cace7fc5b5.

**Pull request merged**